search/src/main/scala/weco/api/search/json/CatalogueJsonUtil.scala

```diff
@@ -18,33 +13,91 @@ trait CatalogueJsonUtil {
-        .removeKeyIf(includes.succeededBy, "succeededBy")
+        .removeKeyRecursivelyIf(!includes.identifiers, "identifiers")
+        .removeKeyIf(!includes.items, "items")
+        .removeKeyIf(!includes.holdings, "holdings")
+        .removeKeyIf(!includes.subjects, "subjects")
+        .removeKeyIf(!includes.genres, "genres")
+        .removeKeyIf(!includes.contributors, "contributors")
+        .removeKeyIf(!includes.production, "production")
+        .removeKeyIf(!includes.languages, "languages")
+        .removeKeyIf(!includes.notes, "notes")
+        .removeKeyIf(!includes.images, "images")
+        .removeKeyIf(!includes.parts, "parts")
+        .removeKeyIf(!includes.partOf, "partOf")
+        .removeKeyIf(!includes.precededBy, "precededBy")
+        .removeKeyIf(!includes.succeededBy, "succeededBy")
         .deepDropNullValues
   }
 
   implicit class ImageOps(im: Image[ImageState.Indexed]) {
     def asJson(includes: MultipleImagesIncludes): Json =
-      DisplayImage(im, includes).asJson
+      DisplayImage(im, includes = MultipleImagesIncludes.all).asJson
+        .withIncludes(includes)
+
+    def asJson(
+      includes: SingleImageIncludes,
+      visuallySimilar: Option[Seq[Image[ImageState.Indexed]]],
+      withSimilarColors: Option[Seq[Image[ImageState.Indexed]]],
+      withSimilarFeatures: Option[Seq[Image[ImageState.Indexed]]]
+    ): Json = {
+      val baseJson =
+        DisplayImage(im, includes = SingleImageIncludes.all).asJson
+          .addImagesIf(
+            includes.visuallySimilar,
+            key = "visuallySimilar",
+            value = visuallySimilar
+          )
+          .addImagesIf(
+            includes.withSimilarColors,
+            key = "withSimilarColors",
+            value = withSimilarColors
+          )
+          .addImagesIf(
+            includes.withSimilarFeatures,
+            key = "withSimilarFeatures",
+            value = withSimilarFeatures
+          )
+
+      baseJson.withIncludes(includes)
+    }
   }
 
   implicit class ImplicitCatalogueJsonOps(j: Json) {
     def removeKeyRecursivelyIf(b: Boolean, key: String): Json =
-      if (!b) j.removeKeyRecursively(key) else j
+      if (b) j.removeKeyRecursively(key) else j
 
     def removeKeyIf(b: Boolean, key: String): Json =
-      if (!b) j.removeKey(key) else j
+      if (b) j.removeKey(key) else j
+
+    def addImagesIf[V](
+      b: Boolean,
+      key: String,
+      value: Option[Seq[Image[ImageState.Indexed]]]
+    ): Json =
+      if (b)
+        j.mapObject(
+          jsonObj =>
+            value match {
+              case Some(v) =>
+                jsonObj.add(
+                  key,
+                  value
+                    .map(
+                      images =>
+                        images.map(DisplayImage(_, SingleImageIncludes.all))
+                    )
+                    .asJson
+                )
+              case None => jsonObj
+            }
+        )
+      else
+        j
```

Comment:
When the images index contains display images, we can modify this method to take `value: Option[Seq[Json]]`, where the JSON objects are whatever we read out of the `"display"` field in the index.